onegov/election_day/formats/election/wabstic_proporz.py

```diff
@@ -69,34 +69,99 @@ def line_is_relevant(line, number, district=None):
         return line.sortgeschaeft == number
 
 
-def get_entity_id(line, entities):
-    entity_id = int(line.bfsnrgemeinde or 0)
-    return 0 if entity_id in EXPATS else entity_id
-
-
-def get_candidate_id(line):
-    return str(int(line.knr))
+def get_number_of_mandates(line):
+    col = 'sitze'
+    if not hasattr(line, col):
+        raise ValueError(_('Missing column: ${col}', mapping={'col': col}))
+    try:
+        return int(line.sitze or 0)
+    except ValueError:
+        raise ValueError(
+            _('Invalid integer: ${col}', mapping={'col': col}))
+
+
+def get_ausmittlungsstand(line):
+    col = 'ausmittlungsstand'
+    try:
+        t = int(line.ausmittlungsstand or 0)
+        assert 0 <= t <= 3
```

Comment:
You shouldn't use assertions and exceptions like this. Exceptions are for unexpected things (a value being invalid is expected in an import). Assertions on the other hand are for developers only. The end-user should never see the result of an assertion.

Here, you could simply use a condition, then raise a ValueError:

```python
if not (0 <= t <= 3): 
   raise ValueError( _('Value ${col} is not between 0 and 3', mapping={'col': col})))
```

Of course you would have to do that outside the current try block (or you would catch your own exception). But that would be clearer anyway and it would make obvious that you are not returning the `t` anywhere in the function :wink:

I would also rename `t` into something a bit more lengthy, like maybe `result`.

Reply:
Yes, agree 100%.